plc4j/core/src/main/java/org/apache/plc4x/java/connectionpool/PlcConnectionManager.java

```diff
@@ -0,0 +1,154 @@
+/*
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+*/
+package org.apache.plc4x.java.connectionpool;
+
+import org.apache.plc4x.java.PlcDriverManager;
+import org.apache.plc4x.java.PlcDriverManagerInterface;
+import org.apache.plc4x.java.api.authentication.PlcAuthentication;
+import org.apache.plc4x.java.api.connection.PlcConnection;
+import org.apache.plc4x.java.api.exceptions.PlcConnectionException;
+import org.apache.plc4x.java.exceptions.NoConnectionAvailableException;
+import org.apache.plc4x.java.exceptions.NotConnectedException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.PreDestroy;
+import java.util.HashMap;
+import java.util.Map;
+
+
+/**
+ * Plc Connection Manager to act as a controller to give threads access to a connection in the pool only if it is not
+ * used by a different thread
+ */
+public class PlcConnectionManager implements PlcDriverManagerInterface {
+    private static final Logger logger = LoggerFactory.getLogger(PlcConnectionManager.class);
+    //TODO: implement connection Pool with more than one connection
+    private Map<String, PlcConnection> connections = new HashMap<>();
```

Comment:
I think we have to use a ConcurrentMap or to use locks to avoid ConcurrentModificationExceptions, as multiple threads could modify the list concurrently.